### PR TITLE
Add context param to emitter.New* and outputflag.Flags.Open

### LINKED ISSUE #1881: Add context.Context parameter to emitter.New*
For #1271, I want to wrap a `csvio.Writer` with a new `FuseWriter` that will send records through a `fuser.Fuser`. Since `Fuser` has to see its entire input before it can produce any output, `FuseWriter` will wait until it's closed before touching the underlying `csvio.Writer`, at which point it will copy the `Fuser`'s buffered records to the `csvio.Writer`.

We want the copy operation to observe `context.Context` cancelation. For the `zq` command, the straightforward way to do that appears to involve threading the `Context` from `signalctx.New` through `outputflags.Flags.Open` into `emitter.New*` and finally to `source.NewWriter`, which already accepts a `Context`.

### PR DESCRIPTION
**Pull request by nwt**
Closes #1881.

@mattnibs: Please be sure to take a look at this one.

### CODE REVIEW

File: ppl/cmd/zar/find/command.go [hunk 1]
```diff
@@ -109,21 +109,21 @@ func (c *Command) Run(args []string) error {
 		findOptions = append(findOptions, archive.SkipMissing())
 	}
 
+	ctx, cancel := context.WithCancel(context.Background())
```
Comment (alfred-landrum) [inline ppl/cmd/zar/find/command.go:112]: nit: make this context.TODO()

**Review by alfred-landrum — APPROVED** [review]

File: ppl/cmd/zar/find/command.go (hunk 1 above)
Comment (nwt) [reply ppl/cmd/zar/find/command.go]: Done in 921f5b6c7c5619c422e75072cf1d8b0f0d78220a.

File: python/brim/src/zqext.go [hunk 2]
```diff
@@ -61,7 +61,7 @@ func doZqlFileEval(inquery, inpath, informat, outpath, outformat string) (err er
 	}
 	defer rc.Close()
 
-	w, err := emitter.NewFile(outpath, zio.WriterOpts{
+	w, err := emitter.NewFile(context.Background(), outpath, zio.WriterOpts{
```
Comment (mattnibs) [inline python/brim/src/zqext.go:64]: Shouldn't this be TODO?
```suggestion
	w, err := emitter.NewFile(context.TODO(), outpath, zio.WriterOpts{
```

**Review by mattnibs — APPROVED** [review]

File: python/brim/src/zqext.go (hunk 2 above)
Comment (nwt) [reply python/brim/src/zqext.go:64]: I used `context.Background` since that's what's used below for the call to `driver.Run`.

**Pull request merged**